plugin/http/propagation/propagation.go

```diff
@@ -0,0 +1,86 @@
+// Copyright 2018, OpenCensus Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Package propagation contains HTTP propagators.
+package propagation
+
+import (
+	"encoding/hex"
+	"fmt"
+	"net/http"
+	"strings"
+
+	"go.opencensus.io/trace"
+)
+
+const (
+	supportedVersion = 0
+	header           = "TraceContext"
+)
+
+// HTTPFormat implements the TraceContext trace propagation format.
+// See https://github.com/w3c/distributed-tracing for more information.
+type HTTPFormat struct{}
+
+// FromRequest extracts a TraceContext span context from incoming requests.
+func (f *HTTPFormat) FromRequest(req *http.Request) (sc trace.SpanContext, ok bool) {
+	h := req.Header.Get(header)
+	if h == "" {
+		return trace.SpanContext{}, false
+	}
+	sections := strings.Split(h, "-")
+	if len(sections) < 3 {
+		return trace.SpanContext{}, false
+	}
+
+	ver, err := hex.DecodeString(sections[0])
+	if err != nil {
+		return trace.SpanContext{}, false
+	}
+	if len(ver) == 0 || int(ver[0]) > supportedVersion {
```

Comment:
Perhaps let's make this version check stricter/more comprehensive.

https://w3c.github.io/distributed-tracing/report-trace-context.html#version
say:
a) is a 1 byte value
b) representing an 8-bit unsigned integer
c) Version 255 is reserved

However, the current supported version seems to only be 0 as you already have in the code and also per https://w3c.github.io/distributed-tracing/report-trace-context.html#version-0.

Maybe my suggestions would be relevant in the future once other versions are supported?

Reply:
Checking the max version.